test/core/end2end/goaway_server_test.c

```diff
@@ -186,9 +186,11 @@ int main(int argc, char **argv) {
   CQ_EXPECT_COMPLETION(cqv, tag(0x301), 1);
   cq_verify(cqv);
 
-  GPR_ASSERT(GRPC_CHANNEL_READY ==
-             grpc_channel_check_connectivity_state(chan, 0));
-  grpc_channel_watch_connectivity_state(chan, GRPC_CHANNEL_READY,
+  /* accept IDLE, CONNECTING or READY */
+  const grpc_connectivity_state connectivity_state =
+      grpc_channel_check_connectivity_state(chan, 0);
+  GPR_ASSERT(connectivity_state <= 2);
```

Comment:
Also: this seems bogus...

We've successfully started an RPC. The channel should be ready.